### PR TITLE
typing in data curation folder and rest of repo

### LINKED ISSUE #579: Typing in data-curation folder and rest of repo
This ticket would complete the work started here: https://app.zenhub.com/workspaces/dibbs-text-to-code-68756bf8ea06bb00319391c0/issues/gh/cdcgov/dibbs-text-to-code/573

by expanding to include data-curation, and working on the exclusion files to have for the rest of the repo. Because there are nearly 300 typing errors to work through there, I opted to split the work into this ticket to prevent the PR from being too unwieldy to review.

### PR DESCRIPTION
**Pull request by robertandremitchell**
## Description

Loads of type fixes to `data/` and `data-curation/` folders. deciding to exclude `/azure_scripts`, `/docs` and any archived given that much of that code was exploratory/spikes and beginning to be incorporated into other parts of the repo.

## Related Issues

Closes #579 

## Additional Notes

[Add any additional context or notes that reviewers should know about.]

<--------------------- REMOVE THE LINES BELOW BEFORE MERGING --------------------->

## Checklist

Please review and complete the following checklist before submitting your pull request:

- [ ] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [ ] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [ ] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.

## Checklist for Reviewers

Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: packages/data-curation/src/data_curation/terminologies/loinc.py [hunk 1]
```diff
@@ -257,7 +257,7 @@ def process_loincs_for_umls_urls() -> dict:
     return umls_loinc_results
 
 
-def get_all_loinc_terms_per_code(loinc_result: dict, loinc_rows: list[dict]) -> dict:
+def get_all_loinc_terms_per_code(loinc_result: dict, loinc_rows: list[dict]) -> list[dict]:
```
Comment (JNygaard-Skylight) [inline packages/data-curation/src/data_curation/terminologies/loinc.py]: Seems like we should create a `LoincRow` dataclass.

File: packages/data-curation/src/data_curation/data_emulation.py [hunk 2]
```diff
@@ -110,19 +111,33 @@
     "by screen method": "Screening",
     "by automated count": "Automated"
 }
-BASE_POST_PROCESSING_OPTIONS = [
+PostProcessingOption = typing.Literal[
```
Comment (JNygaard-Skylight) [inline packages/data-curation/src/data_curation/data_emulation.py]: Tiny nit: In other modules  are not qualifying `typing` imports and are importing them directly.

Comment (JNygaard-Skylight) [inline packages/data-curation/src/data_curation/data_emulation.py]: I'm not sure but I think this might be cleaner if this as a `strEum` instead of a list of string literals. I'm not confident about it though.

File: packages/data-curation/src/data_curation/terminologies/loinc.py (hunk 1 above)
Comment (robertandremitchell) [reply packages/data-curation/src/data_curation/terminologies/loinc.py]: updated. I may create a follow-up ticket about reorganizing some of this code, because there are a lot of dataclasses that should probably live in a `models/` folder, but I'm not sure how much that reorg should be in this ticket.

File: packages/data-curation/src/data_curation/data_emulation.py (hunk 2 above)
Comment (robertandremitchell) [reply packages/data-curation/src/data_curation/data_emulation.py]: updated!

File: packages/data-curation/src/data_curation/terminologies/hl7.py [hunk 3]
```diff
@@ -83,7 +83,8 @@ def get_hl7_encounter_act_codes() -> list[dict]:
         hl7_rows.append(external_encounter)
         # TODO: In Subsequent PR update this to be a logging statement
         print(f"HL7 Encounter Act Codes Retrieved from HL7 Act Codes: {len(hl7_rows)}")
-        return hl7_rows
+
+    return hl7_rows
```
Comment (BradySkylight) [inline packages/data-curation/src/data_curation/terminologies/hl7.py:87]: Oops... nice find and fix!

File: packages/data-curation/src/data_curation/terminologies/loinc.py (hunk 1 above)
Comment (BradySkylight) [reply packages/data-curation/src/data_curation/terminologies/loinc.py]: I had considered doing that, but not sure we want to go that deep down the rabbit hole for the results from LOINC.

File: packages/data-curation/src/data_curation/terminologies/loinc.py [hunk 4]
```diff
@@ -368,7 +368,9 @@ def get_loinc_current_version_data() -> tuple[str, str]:
         print(
             f"ERROR Retrieving LOINC META Data for current Version: {loinc_response.status_code}: {loinc_response.text}"
         )
-        return None
+        raise RuntimeError(
+            f"ERROR Retrieving LOINC META Data for current Version: {loinc_response.status_code}: {loinc_response.text}"
+        )
```
Comment (BradySkylight) [inline packages/data-curation/src/data_curation/terminologies/loinc.py:375]: Nice!

File: packages/data-curation/src/data_curation/terminologies/maintenance.py [hunk 5]
```diff
@@ -20,6 +20,8 @@ def update_loinc_embeddings():
     loinc_version, loinc_version_date = get_loinc_current_version_data()
     # find the existing TTC LOINC LabNames file to use for comparison
     current_loinc_file = get_latest_extract_file_name(LAB_NAMES)
+    if current_loinc_file is None:
+        return
```
Comment (BradySkylight) [inline packages/data-curation/src/data_curation/terminologies/maintenance.py:24]: good catch!

File: packages/data-curation/src/data_curation/terminology_extraction.py
Comment (BradySkylight) [inline packages/data-curation/src/data_curation/terminology_extraction.py:1]: Just note that this was left as a remnant of the full extraction process for all terminology sets and may not be used again, as we are leveraging the maintenance/update process going forward.

File: packages/data-curation/src/data_curation/terminology_extraction.py [hunk 6]
```diff
@@ -262,12 +263,12 @@ def process_loinc_codes_with_umls(file_path: str) -> dict:  # noqa: D103
...
 
 
-def save_valueset_csv_file(filename: str, contents: dict, append_to_file: bool = False):  # noqa: D103
+def save_valueset_csv_file(filename: str, contents: list[dict] | None, append_to_file: bool = False):  # noqa: D103
     if not filename.strip():
         print("No filename supplied.  Failed to save CSV file!")
         return
 
-    if contents is None and len(contents) == 0:
+    if contents is None or len(contents) == 0:
```
Comment (BradySkylight) [inline packages/data-curation/src/data_curation/terminology_extraction.py:271]: nice

**Review by BradySkylight — APPROVED** [review]
Nice!  If this merges before my PR, I'll likely have a few merge conflicts to deal with.  But glad to see the typing and changes!  Thanks Rob!

File: packages/data-curation/src/data_curation/terminologies/loinc.py (hunk 1 above)
Comment (robertandremitchell) [reply packages/data-curation/src/data_curation/terminologies/loinc.py]: yeah, it's a question I had while looking through validation too of how much we should get `validation/` and in this case `data-curation/` into a similar shape as the other repos. Definitely not a high priority.

File: packages/data-curation/src/data_curation/terminology_extraction.py
Comment (robertandremitchell) [reply packages/data-curation/src/data_curation/terminology_extraction.py:1]: oh, good call out. data-curation is admittedly the part of the repo I'm weakest in, so if there's more code that should get moved off to archived, lmk!

File: packages/data-curation/src/data_curation/data_emulation.py (hunk 2 above)
Comment (robertandremitchell) [reply packages/data-curation/src/data_curation/data_emulation.py]: I think this makes sense in a bigger refactor of this module where something like this lives in a `models` folder to better import/manage into `post_process.py` and other places. not opposed to doing it here, there's definitely a few spots in the code I think could be better organized/refactored, but I'm admittedly not confident enough in how all this code works together to do it here.
